compiler/rustc_middle/src/mir/interpret/mod.rs

```diff
@@ -49,7 +49,7 @@ macro_rules! err_exhaust {
 #[macro_export]
 macro_rules! err_machine_stop {
     ($($tt:tt)*) => {
-        $crate::mir::interpret::InterpError::MachineStop(Box::new($($tt)*))
+        $crate::mir::interpret::InterpError::MachineStop(box ($($tt)*))
```

Comment:
This will mean other crates using the macro (such as standalone Miri) need to also enable the `box` feature, right? I think I'd prefer to avoid that.

Reply:
There exists `#[allow_internal_unstable]` to prevent that.

Reply:
Example macro that does this for box syntax is the [vec macro](https://doc.rust-lang.org/1.54.0/src/alloc/macros.rs.html#42). As @petrochenkov correctly assumed, this PR was a test run though to explore the implications of using box syntax everywhere.